auth.go

```diff
@@ -0,0 +1,132 @@
+package sqlite
+
+// #include <stdint.h>
+// #include <sqlite3.h>
+// extern int go_sqlite_auth_tramp(uintptr_t, int, char*, char*, char*, char*);
+// static int c_auth_tramp(void *userData, int action, const char* arg1, const char* arg2, const char* db, const char* trigger) {
+//   return go_sqlite_auth_tramp((uintptr_t)userData, action, (char*)arg1, (char*)arg2, (char*)db, (char*)trigger);
+// }
+// static int sqlite3_go_set_authorizer(sqlite3* conn, uintptr_t id) {
+//   return sqlite3_set_authorizer(conn, c_auth_tramp, (void*)id);
+// }
+import "C"
+import (
+	"errors"
+	"sync"
+)
+
+// An Authorizer is called during statement preparation to see whether an action
+// is allowed by the application. See https://sqlite.org/c3ref/set_authorizer.html
+type Authorizer interface {
+	Authorize(action OpType, info ActionInfo) AuthResult
+}
+
+// ActionInfo holds information about an action to be authorized.
+type ActionInfo struct {
+	Arg1     string
+	Arg2     string
+	Database string
+	Trigger  string
+}
```

Comment:
I'm going to make this type more specific and useful to Go devs by removing Arg1 and Arg2 and replacing them with name specific types that match the SQLite documentation such as Index, Table. View, and then having the trampoline populate the relevant values based on the action.

Reply:
As such it makes sense to merge the OpType var to this struct.